source/ada/lsp-ada_documents.adb

```diff
@@ -2047,50 +2043,68 @@ package body LSP.Ada_Documents is
                      Name_Vectors.Empty_Vector,
                      Cursor,
                      Inserted);
-               end if;
 
-               Self.Symbol_Cache (Cursor).Append
-                 ((Node.As_Defining_Name,
-                   Global_Visible.Unchecked_Get.Evaluate (Node)));
+                  Self.Symbol_Cache (Cursor).Append
+                    ((Node.As_Defining_Name,
+                     Global_Visible.Unchecked_Get.Evaluate (Node)));
+               end if;
             end;
          end loop;
       end Refresh_Symbol_Cache;
 
-      Cursor : Symbol_Maps.Cursor;
+      procedure Insert (Item : Name_Information) is
+      begin
+         if not Result.Contains (Item.Name) and then
+           (not Only_Public or else Item.Is_Public)
+         then
+            Result.Insert
+              (Item.Name,
+               (Is_Dot_Call  => False,
+                Is_Visible   => False,
+                Use_Snippets => False,
+                Pos          => <>));
+         end if;
+      end Insert;
 
+      Cursor : Symbol_Maps.Cursor;
    begin
       if Self.Refresh_Symbol_Cache then
          Refresh_Symbol_Cache;
          Self.Refresh_Symbol_Cache := False;
       end if;
 
-      Cursor := Self.Symbol_Cache.Ceiling (Prefix);
+      if Pattern.Get_Kind = LSP.Messages.Full_Text
+        and then not Pattern.Get_Negate
+        and then Pattern.Get_Whole_Word
```

Comment:
Do we really need Whole_Word in the condition? Before this PR we did `Ceiling` for ordinary search (that isn't Whole_Word, if I understand it right).